# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../l7-filter/last_lx26.diff
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- ./kernel-2.6-layer7-1.0.patch.orig	2005-03-09 17:54:06.272347280 -0300
+++ ./kernel-2.6-layer7-1.0.patch	2005-03-09 17:58:44.443058904 -0300
@@ -99,9 +99,9 @@
 diff -Nurp linux-2.6.10-stock/net/ipv4/netfilter/ip_conntrack_core.c linux-2.6.10-layer7-clean/net/ipv4/netfilter/ip_conntrack_core.c
 --- linux-2.6.10-stock/net/ipv4/netfilter/ip_conntrack_core.c	2004-12-24 15:33:47.000000000 -0600
 +++ linux-2.6.10-layer7-clean/net/ipv4/netfilter/ip_conntrack_core.c	2005-01-07 13:26:59.000000000 -0600
-@@ -302,6 +302,14 @@ destroy_conntrack(struct nf_conntrack *n
- 	if (ct->expecting)
- 		remove_expectations(ct, 1);
+@@ -253,6 +253,14 @@
+ 		list_del(&ct->tuplehash[IP_CT_DIR_ORIGINAL].list);
+ 	}
  
 +        #if defined(CONFIG_IP_NF_MATCH_LAYER7) || defined(CONFIG_IP_NF_MATCH_LAYER7_MODULE)
 +        /* This ought to get free'd somewhere.  How about here? */
@@ -111,9 +111,9 @@
 +                kfree(ct->layer7.app_data);
 +        #endif
 +
- 	/* Delete our master expectation */
- 	if (ct->master) {
- 		if (ct->master->expectant) {
+ 	CONNTRACK_STAT_INC(delete);
+ 	WRITE_UNLOCK(&ip_conntrack_lock);
+ 
 diff -Nurp linux-2.6.10-stock/net/ipv4/netfilter/ip_conntrack_standalone.c linux-2.6.10-layer7-clean/net/ipv4/netfilter/ip_conntrack_standalone.c
 --- linux-2.6.10-stock/net/ipv4/netfilter/ip_conntrack_standalone.c	2004-12-24 15:34:33.000000000 -0600
 +++ linux-2.6.10-layer7-clean/net/ipv4/netfilter/ip_conntrack_standalone.c	2005-01-07 13:26:59.000000000 -0600
